docs/reference/protocol-spec.md

```diff
@@ -85,13 +86,23 @@ Redis generally uses RESP as a [request-response](#request-response-model) proto
 * The server replies with a RESP type.
   The reply's type is determined by the command's implementation and possibly by the client's protocol version.
 
-We categorize every RESP data type as either _simple_ or _aggregate_.
-Simple types are similar to scalars in programming languages that represent plain literal values, for example, Booleans.
-Aggregates, such as Arrays and Maps, can have varying numbers of sub-elements and nesting levels.
+RESP is **always** ASCII-encoded (UTF-8/Latin-1).
+
+The `\r\n` (CRLF) is the protocol's _terminator_, which **always** separates its parts.
 
-The first byte in an RESP-serialized payload identifies its type.
+The first byte in an RESP-serialized payload always identifies its type.
 Subsequent bytes constitute the type's contents.
-The `\r\n` (CRLF) is the protocol's _terminator_, which always separates its parts.
+
+We categorize every RESP data type as either _simple_, _bulk_ or _aggregate_.
+
+Simple types are similar to scalars in programming languages that represent plain literal values. Booleans and Integers are such examples.
+
+RESP strings are either _simple_ or _bulk_.
+Simple strings never contain carriage return (`\r`) or line feed (`\n`) characters.
+Bulk strings can contain any ASCII character and may also be referred to as _binary_ or _blob_.
```

Comment:
Apart from ASCII characters (bytes in the range 0-127), it may also contain the bytes values 128-255. I think it would be better to talk about binary data than to talk about character encodings here.

Reply:
👍 So what is the proper name for 8-bit ASCII? "Extended"? "winblows-1252"? "ISO 8859-1"?

Reply:
ASCII is 7-bit, never 8-bit. Uninterpreted octets (a.k.a. bytes) are not characters (neither ASCII nor Latin 1 nor UTF-8). So, I agree with @zuiderkwast, it is better to never mention "character" (unless a specific encoding is also mentioned), and for raw data it is better to talk about binary bytes.

Of course, for convenience a binary protocol can still use sequences of binary bytes whose values happen to form meaningful English words when interpreted as 7-bit ASCII but the protocol is then still binary, and the bytes are still raw bytes.

Reply:
Telling that bulk strings should either (choose the right one):
- neither contain <CR> or <LF>
- not contain the sequence <CR><LF>
is enough, IMHO

Reply:
@zuiderkwast I think he meant simple string.
@zuiderkwast some decoders might check only for CR (and assume there is LF right after it), I think "neither CR nor LF" is safer.